feat: use autowire for configuration (#2112)
shopware/docs: guides/plugins/plugins/framework/data-handling/replacing-associated-data.md

In order to delete it, we once again need its repository. The name for the entity can be found in the definition, to be precise inside of the `getEntityName` method.

> So let's inject this repository into our class called `ReplacingData`. With `autowire` enabled in your `services.php`, the repositories are automatically injected into the constructor — no additional configuration is needed.

Reviewer: ```suggestion
With `autowire` enabled in your `services.php`, the repositories are automatically injected into the constructor of the class called `ReplacingData`.
```
